h/dsutils.h

```diff
@@ -24,40 +24,118 @@ static int defaultCSIFieldCount = 3;
+ */
 typedef struct Volser_tag {
   char value[6]; /* space-padded */
 } Volser;
 
-int getVolserForDataset(const DatasetName *dataset, Volser *volser);
-char getRecordLengthType(char *dscb);
-int getMaxRecordLength(char *dscb);
-
-int obtainDSCB1(const char *dsname, unsigned int dsnameLength,
-                       const char *volser, unsigned int volserLength,
-                       char *dscb1);
-int getLreclOrRespondError(HttpResponse *response,
-		const DatasetName *dsn,
-		const char *ddPath);
-bool isDatasetPathValid(const char *path);
-
-void extractDatasetAndMemberName(const char *datasetPath,
-                                        DatasetName *dsn,
-                                        DatasetMemberName *memberName);
-void respondWithDYNALLOCError(HttpResponse *response,
-                              int rc, int sysRC, int sysRSN,
+/**
+ * @brief Gets the serial volume of a dataset.
+ *
+ * @param dataset dataset to be called when the function is
+ * invoked. To be validated first and get its value for memcpy.
+ * @param volser To pass its value and to be used in memcpy.
+ * @return 0 is returned when the function has no error.
+ * -1 is returned when an error occurred 
+ */
+int dsutilsGetVolserForDataset(const DatasetName *dataset, Volser *volser);
+
+/**
+ * @brief Gets the record format of a dataset.
+ *
+ * @param dscb a parameter to get the details of the physical
+ * characteristics of a dataset.
+ * @return U is returned when Undefined-length dataset.
+ * F is returned when it is a fixed block and V for variable block.
+ */
+char dsutilsGetRecordLengthType(const char *dscb);
+
+/**
+ * @brief Gets the max record length of a dataset.
+ *
+ * @param dscb a parameter to get the details of the physical
+ * characteristics of a dataset.
+ * @return lrecl is the maximum length of a dataset .
+ */
+int dsutilsGetMaxRecordLength(const char *dscb);
+
+/**
+ * @brief Gets the Data Set Control Block of a dataset.
+ *
+ * @param dsname a parameter to get the dataset name.
+ * @param dsnamelength a parameter to get the dataset length.
+ * @param volser a parameter to get dataset the volume serial.
+ * @param volserlength a parameter to get the volume serial length.
+ * @param dscb1 a parameter to get the details of the physical
+ * characteristics of a dataset.
+ * @return 0 is successful obtaining the dscb.
+ */
+int dsutilsObtainDSCB1(const char *dsname, unsigned int dsnameLength, const char *volser, unsigned int volserLength, char *dscb1);
+
+/**
+ * @brief Gets the length of a dataset or respond an error.
+ *
+ * @param response a parameter to get a response when this 
+ * function encountered an error
+ * @param dsn a parameter to get the dataset name
+ * @param ddpath a parameter to get dataset path
+ * @return lrecl return the length of a dataset 
+ * 0 this function encountered error 
+ */
+int dsutilsGetLreclOrRespondError(HttpResponse *response, const DatasetName *dsn, const char *ddPath);
```

Comment:
The "respond" functions and the functions with http related code don't seem to belong here. 

The name `dsutils` implies some generic data set utils, but the source contains http service speicific code . Did you intend that?

Reply:
Updated dustils and jesService.